woocommerce/woocommerce-android
# 9352 [Product discount] Order Create > Product Details refactoring

Implements part of #9352


### Description

This PR implements foundations for direct product discounts feature. It brings the following changes:
1. Refactored `OrderCreateEditProductDetailsFragment`:
- Removed `sharedViewModel` reference  🎉
- Introduced `Fragment Result API` to pass the result back to the `OrderCreateEditViewModel`
- Created `OrderCreateEditProductDetailsViewModel` and moved the view state computation logic from the fragment to there
2. Introduced `FeatureFlag.ORDER_CREATION_PRODUCT_DISCOUNTS` and hid product discount - related UI components under it.
3. Added empty screen - `OrderCreateEditProductDiscountFragment` and `OrderCreateEditProductDiscountViewModel`
4. Reorganized packages

The following PR(s) will bring a product discount feature implementation and unit tests.

### Testing instructions

When running the release build (`ORDER_CREATION_PRODUCT_DISCOUNTS` = false) there should be no visual changes.
Make sure the `Order > Product Details` screen works correctly:
1. Try to remove the item using product details screen
2. Verify "back press" and "x" navigation icon redirect back as expected
5. Verify no other regression happens

### Images/gif

No visual changes are expected.
